qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1511B.py

```diff
@@ -0,0 +1,110 @@
+        super().__init__(parent, name, slot_nr, **kwargs)
+        self._valid_i_measure_ranges: List[IMeasRange] = [IMeasRange.AUTO,
+                                                          IMeasRange.MIN_1nA,
+                                                          IMeasRange.MIN_10nA,
+                                                          IMeasRange.MIN_100nA,
+                                                          IMeasRange.MIN_1uA,
+                                                          IMeasRange.MIN_10uA,
+                                                          IMeasRange.MIN_100uA,
+                                                          IMeasRange.MIN_1mA,
+                                                          IMeasRange.MIN_10mA,
+                                                          IMeasRange.MIN_100mA,
+                                                          IMeasRange.FIX_1nA,
+                                                          IMeasRange.FIX_10nA,
+                                                          IMeasRange.FIX_100nA,
+                                                          IMeasRange.FIX_1uA,
+                                                          IMeasRange.FIX_10uA,
+                                                          IMeasRange.FIX_100uA,
+                                                          IMeasRange.FIX_1mA,
+                                                          IMeasRange.FIX_10mA,
+                                                          IMeasRange.FIX_100mA]
+        self._valid_v_output_ranges: List[VOutputRange] = [
+            VOutputRange.AUTO, VOutputRange.MIN_0V5, VOutputRange.MIN_2V,
+            VOutputRange.MIN_5V, VOutputRange.MIN_20V, VOutputRange.MIN_40V,
+            VOutputRange.MIN_100V]
+        self._valid_i_output_ranges: List[IOutputRange] = [
+            IOutputRange.AUTO, IOutputRange.MIN_1nA, IOutputRange.MIN_10nA,
+            IOutputRange.MIN_100nA, IOutputRange.MIN_1uA,
+            IOutputRange.MIN_10uA, IOutputRange.MIN_100uA,
+            IOutputRange.MIN_1mA, IOutputRange.MIN_10mA, IOutputRange.MIN_100mA]
+
+        if asu_present:
+            self._valid_i_measure_ranges = self._valid_i_measure_ranges + \
+                                           [IMeasRange.MIN_1pA,
+                                            IMeasRange.MIN_10pA,
+                                            IMeasRange.MIN_100pA,
+                                            IMeasRange.FIX_1pA,
+                                            IMeasRange.FIX_10pA,
+                                            IMeasRange.FIX_100pA]
+            self._valid_i_output_ranges = self._valid_i_output_ranges + \
+                                          [IOutputRange.MIN_1pA,
+                                           IOutputRange.MIN_10pA,
+                                           IOutputRange.MIN_100pA]
+
+    def source_config(
```

Comment:
why can't this method belong to the parent class? and B1511B will then just define the valid_* lists (as it already does it ints init) without the need for modifying the source_config method?